File: Baekjoon/Level7/7-05/main.c
#include <stdio.h>
#define MAX_STRLEN 1000001

int main() {
    char str[MAX_STRLEN];
    int i, alphabet[26] = {0,};
    fgets(str, sizeof(str), stdin);
    for (i = 0; str[i] != '\0'; i++) {
        if (str[i] >= 'a' && str[i] <= 'z')
            str[i] = str[i] - 32;   // 대문자로 변경
        alphabet[str[i]-65]++;
    }
    int max = 0; char max_alpha = 0;
    for (i = 0; i < 26; i++) {
        if (alphabet[i] > max) {
            max = alphabet[i];
            max_alpha = i;
        }
        else if (max == alphabet[i])
            max_alpha = '?';
    }
    if (max_alpha == '?')
        printf("%c", max_alpha);
    else
        printf("%c", max_alpha + 65);
    return 0;
}

File: Baekjoon/Level7/7-05/main.dart
import 'dart:io';

String solution(String input) {
  String answer = '';
  List<int> alphabet = List.filled(26, 0);
  input = input.toUpperCase();
  for (var s in input.split('')) {
    // 문자를 아스키 코드로 변환후 인덱스화
    int index = s.codeUnitAt(0) - 'A'.codeUnitAt(0);
    alphabet[index]++;
  }
  int max = 0;
  for (int i = 0; i < alphabet.length; i++) {
    if (alphabet[i] == 0) continue;
    if (max == alphabet[i]) {
      answer = '?';
    }
    if (max < alphabet[i]) {
      max = alphabet[i];
      answer = String.fromCharCode(i + 'A'.codeUnitAt(0)); // 아스키 코드를 문자로 변경
    }
  }
  return answer;
}

void main() {
  String input = stdin.readLineSync()!;

  print(solution(input));
}
